src/libproc_macro/lib.rs

```diff
@@ -95,6 +96,12 @@ impl TokenStream {
     }
 }
 
+/// Attempts to break the string into tokens and parse those tokens into a token stream.
+/// May fail for a number of reasons, for example, if the string contains unbalanced delimiters
+/// or characters not existing in the language.
+///
+/// REVIEW The function actually panics on any error and never returns `LexError`.
```

Comment:
I think this doesn't match the original implementation, when we first started we were sure to return an error I thought...

In any case I'd think it's a bug that this panics, it should definitely be returning an error (but we can fix that later)